azimuth/modules/dataset_analysis/dataset_warnings.py

```diff
@@ -193,135 +195,117 @@ def get_general_warnings(
-        )
-
-        # Get mean and std token lengths
-        train_desc = train_df.groupby("label").agg(["mean", "std"])
-        eval_desc = eval_df.groupby("label").agg(["mean", "std"])
-        # Compute divergence
-        divergence = (train_desc - eval_desc)[DatasetColumn.word_count].apply(np.abs)
-        divergence = fill_at_missing_indices(divergence, find_missing_rows(divergence, cls_index))
-        divergence_mean = divergence["mean"].to_numpy()
-        divergence_std = divergence["std"].to_numpy()
-
-        # Make alerts from values.
-        max_delta_mean_words = int(self.config.dataset_warnings.max_delta_mean_words)
-        max_delta_std_words = int(self.config.dataset_warnings.max_delta_std_words)
-        alert_norm_mean = divergence_mean > max_delta_mean_words
-        alert_norm_std = divergence_std > max_delta_std_words
-        alert_norm = np.any([alert_norm_mean, alert_norm_std], axis=0)
+        class_index = np.arange(dm.get_num_classes(labels_only=True))
+        value_per_agg_per_split: Dict[DatasetSplitName, Dict[Agg, float]] = defaultdict(dict)
+        value_per_label_per_agg_per_split: Dict[
+            DatasetSplitName, Dict[Agg, np.ndarray]
+        ] = defaultdict(dict)
+        hist_per_label_per_split = {}
+
+        for split in self.available_dataset_splits:
+            ds = self.get_dataset_split(split).remove_columns([])
+            df = ds.remove_columns(
+                [
+                    column
+                    for column in ds.column_names
+                    if column not in [DatasetColumn.word_count, self.config.columns.label]
+                ]
+            ).to_pandas()
+
+            value_per_agg_per_split[split][Agg.mean] = df["word_count"].mean()
+            value_per_agg_per_split[split][Agg.std] = df["word_count"].std()
+
+            # Get mean and std word count per label
+            stats_per_label = df.groupby("label").agg(["mean", "std"]).reindex(class_index)
+            for agg in [Agg.mean, Agg.std]:
+                value_per_label_per_agg_per_split[split][agg] = stats_per_label["word_count"][
+                    agg
+                ].to_numpy()
+
+            # Get word count histogram per label. Columns are word counts and rows are labels.
+            hist_per_label_per_split[split] = (
+                df.groupby(["label", "word_count"])
+                .size()
+                .unstack(fill_value=0)
+                .reindex(class_index, fill_value=0)
+            )
 
-        # Get counts, this will have N columns with count and LABEL rows.
-        train_count = (
-            train_df.groupby(["label", DatasetColumn.word_count]).size().unstack(fill_value=0)
-        )
-        eval_count = (
-            eval_df.groupby(["label", DatasetColumn.word_count]).size().unstack(fill_value=0)
+        # Compute divergence and alerts
+        divergence_per_label_per_agg = dict()
+        alert_per_label_per_agg = dict()
```

Comment:
Nitpick, but for consistency, it seems that we use `{}` more frequently than the `dict()` in our repo.
```suggestion
        divergence_per_label_per_agg = {}
        alert_per_label_per_agg = {}
```